### PR TITLE
fix: removes all handlers from the root logger

### PR DESCRIPTION
**Pull request by tshu-w**
We should not mutate (insert/remove elements) the list you're currently iterating on. In fact, this causes handlers to not be completely deleted on my side.

For more discussion, see https://stackoverflow.com/questions/7484454/removing-handlers-from-pythons-logging-loggers

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Added links to source code in docs.
 - Fixed issue with GradientDescentTrainer when constructed with validation_data_loader==None and learning_rate_scheduler!=None.
+- Fixed issue with removing all handlers in root logger.
```
Comment (epwalsh) [inline CHANGELOG.md:15]: This should be under the `### Fixed` section (you may need to create this section if it doesn't exist under the `## Unreleased` section.

**Review by epwalsh — CHANGES_REQUESTED** [review]
Thanks @tshu-w, great catch! Just need one small tweak to the CHANGELOG and then we can get this merged.

File: CHANGELOG.md (hunk 1 above)
Comment (tshu-w) [reply CHANGELOG.md]: @epwalsh Does it look okay now? BTW, I've also moved the above fixed to this new section.

**Review by epwalsh — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**